deploy/helm/kubecf/values.schema.yaml

```diff
@@ -0,0 +1,339 @@
+"$schema": http://json-schema.org/draft-07/schema
+"$id": https://raw.githubusercontent.com/cloudfoundry-incubator/kubecf/master/deploy/helm/kubecf/values.schema.yaml
+
+additionalProperties: false
+
+required:
+- system_domain
+
+definitions:
+
+  operation:
+    type: object
+    oneOf:
+    - properties:
+        type: {enum: [replace]}
+        path: {type: string}
+        value: {}
+      required: [type, path, value]
+      additionalProperties: false
+    - properties:
+        type: {enum: [remove]}
+        path: {type: string}
+      required: [type, path]
+      additionalProperties: false
+    - properties:
+        # qcopy and qmove are quarks extensions to bosh operations
+        type: {enum: [qcopy, qmove]}
+        path: {type: string}
+        from: {type: string}
+      required: [type, path, from]
+      additionalProperties: false
+
+  port:
+    type: integer
+    # should minimum be 1?
+    minimum: 1024
+    maximum: 65535
+
+  release:
+    type: object
+    properties:
+      buildpack: {type: string}
+      condition:
+        anyOf:
+        - type: boolean
+        - type: string
+      image:
+        type: object
+        properties:
+          repository: {type: string}
+          tag: {type: string}
+        required: [repository]
+        additionalProperties: false
+      stemcell:
+        type: object
+        properties:
+          os: {type: string}
+          version: {type: string}
+        additionalProperties: false
+      url: {type: string}
+      # XXX only allowed to be null for cflinux release with just a condition
+      version: {"$ref": "#/definitions/string_or_null"}
+    additionalProperties: false
+
+  release_any:
+    anyOf:
+    - "$ref": "#/definitions/release_bosh"
+    - "$ref": "#/definitions/release_native"
+
+  release_bosh:
+    # A bosh release must have a version property, or be disabled
+    allOf:
+    - "$ref": "#/definitions/release"
+    - anyOf:
+      - required: [version]
+      - required: [condition]
+        properties:
+          condition: {enum: [false]}
+
+  release_defaults:
+    # Only valid defaults are "stemcell" and "url", at least one of them being required
+    allOf:
+    - "$ref": "#/definitions/release"
+    - properties:
+        stemcell: {}
+        url: {}
+      additionalProperties: false
+      anyOf:
+      - required: [stemcell]
+      - required: [url]
+
+  release_native:
+    # A native release must have an image property, or be disabled
+    allOf:
+    - "$ref": "#/definitions/release"
+    - anyOf:
+      - required: [image]
+      - required: [condition]
+        properties:
+          condition: {enum: [false]}
+
+  service:
+    type: object
+    properties:
+      annotations:
+        anyOf:
+        - type: object
+          additionalProperties: {type string}
+        - type: 'null'
+      type:
+        type: string
+        enum: [ClusterIP, NodePort, LoadBalancer]
```

Comment:
I think we do, but I didn't have a comprehensive set of rules, and I think the templates will happily add a `clusterIP` to `LoadBalancer`.

If you can provide me a list of constraints, I will add them in this PR. Otherwise I'll leave it for "later"...